app/src/main/kotlin/com/wire/android/ui/calling/incoming/IncomingCallViewModel.kt

```diff
@@ -78,14 +80,16 @@ class IncomingCallViewModel @Inject constructor(
         }
     }
 
-    private fun observeConversationDetails(conversationDetails: ConversationDetails) {
-        val conversationName = when (conversationDetails) {
-            is ConversationDetails.Group -> conversationDetails.conversation.name
-            is ConversationDetails.OneOne -> conversationDetails.otherUser.name
-            else -> null
+    private fun initializeScreenState(conversationDetails: ConversationDetails) {
+        callState = when (conversationDetails) {
+            is ConversationDetails.Group -> callState.copy(conversationName = conversationDetails.conversation.name)
+            is ConversationDetails.OneOne -> {
+                callState.copy(
+                    conversationName = conversationDetails.otherUser.name,
+                    avatarAssetId = conversationDetails.otherUser.completePicture?.let { UserAvatarAsset(it) }
+                )
+            }
+            else -> throw IllegalStateException("Invalid conversation type")
```

Comment:
Maybe substitute for `Self` and be exhaustive, so that if more conversation types are added in the future we can get a warning to handle them here?